src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectAssetFileWatcher.cs

```diff
@@ -48,19 +54,19 @@ internal class ProjectAssetFileWatcher : OnceInitializedOnceDisposed, IVsFileCha
         /// <summary>
         /// Called on project load.
         /// </summary>
-        [ConfiguredProjectAutoLoad(RequiresUIThread = true)]
+        [ConfiguredProjectAutoLoad(RequiresUIThread = false)]
         [AppliesTo(ProjectCapability.CSharpOrVisualBasicOrFSharp)]
         internal void Load()
         {
-            EnsureInitialized();
+            _task = InitializeAsync(_cancellationTokenSource.Token);
         }
 
         /// <summary>
         /// Initialize the watcher.
         /// </summary>
-        protected override void Initialize()
+        protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
         {
-            _fileChangeService = _serviceProvider.GetService<IVsFileChangeEx, SVsFileChangeEx>();
+            _fileChangeService = (IVsFileChangeEx)(await AsyncServiceProvider.GlobalProvider.GetServiceAsync(typeof(SVsFileChangeEx)).ConfigureAwait(false));
```

Comment:
Doesn't this ConfigureAwait defeats the purpose of the change? This says "if I switched to the UI thread, stay on it".

Reply:
In practice, I don't see the method continuing on the UI thread.  On the other hand, I could either change it to true (but am I guaranteed to to not be on the UI thread?), or else explicitly switch to a background thread here.  Thoughts?

Reply:
If it was me, I'd be explicit here. Either with ConfigureAwait(true) or an explicit switch. Either way, document it.

Reply:
I'm guessing IVsFileChangeEx is free-threaded hence why it's stays on the same thread?

Reply:
Your tests are failing, I'm guessing GlobalProvider is null here. I'd follow the pattern that I pointed you to by having another constructor that takes AsyncServiceProvider as a parameter and you pass it from the ImportingConstructor.

Reply:
Tests fixed, and comment and explicit switch added.